platform/mbed_wait_api_no_rtos.c

```diff
@@ -41,3 +43,94 @@ void wait_us(int us)
 
 #endif // #ifndef MBED_CONF_RTOS_PRESENT
 
+// This wait_ns is used by both RTOS and non-RTOS builds
+
+#ifdef __CORTEX_M
+#if (__CORTEX_M == 0 && !defined __CM0PLUS_REV) || __CORTEX_M == 1
+// Cortex-M0 and Cortex-M1 take 7 cycles per iteration - SUBS = 1, 2xNOP = 2, BCS = 3
+#define LOOP_SCALER 6000
+#elif (__CORTEX_M == 0 && defined __CM0PLUS_REV) || __CORTEX_M == 3 || __CORTEX_M == 4 || \
+      __CORTEX_M == 23 || __CORTEX_M == 33
+// Cortex-M0+, M3, M4, M23 and M33 take 6 cycles per iteration - SUBS = 1, 3xNOP = 2, BCS = 2
+// TODO - check M33
```

Comment:
Could we resolve the TODO before committing this?

Reply:
We don't actually have any M33 targets yet, do we? So don't think I've got anything I can test this on.

My belief is that the M33 is basically an M4 in terms of internal architecture, but no firm evidence. Could leave the `__CORTEX_M == 33` case out and deal with it when we have the first actual target.